### PR TITLE
[luci] Introduce TestIsOGraph for connection test

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce TestIsOGraph class for connection test providing
arbitrary N input with template.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #6596, from draft #6597

File: compiler/luci/partition/src/ConnectNode.test.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+namespace luci
+{
+namespace test
+{
+
+template <unsigned N> class TestIsOGraph : public TestIsGraphlet<N>, public TestOGraphlet
+{
+public:
+  TestIsOGraph() = default;
```
Comment (mhs4670go) [inline compiler/luci/partition/src/ConnectNode.test.h:37]: Removing the explicit default constructor is only applied to IR(#6624)?

Comment (seanshpark) [reply compiler/luci/partition/src/ConnectNode.test.h:37]: Yup :)

Comment (seanshpark) [reply compiler/luci/partition/src/ConnectNode.test.h:37]: For some tests graphs there will be empty class if this default ctor is removed. this is OK in syntax but may look a bit difficult to read the code.

**Review by mhs4670go — APPROVED** [review]
LGTM

**Pull request merged**